### ISSUE TITLE
Infer types transitively and allow to ignore false positives

### ISSUE THREAD

**Issue by pavelpatrin** [go-simpler/sloglint#54]
* Infer types dynamically via intermediate containers when this is possible.
* Allow to ignore false positive matches.

```
	// Prepare component tags.
	component := []any{
		slog.String("component", "gRPCServer"),
		slog.String("interceptor", "Auth"),
	}

	// Authenticate by metadata values.
	logger.Debug("Authenticating request", component...)
```

```
auth.go:55:2: key-value pairs should not be used (sloglint)
	logger.Debug("Authenticating request", component...)
````

**Comment by pavelpatrin** [comment]
This is simplified code. Of course I know about `logger.With()` but in my case it is not possible.

**Comment by tmzane** [comment]
I believe this was a problem with unpacked slices and it was fixed in #51. What version are you using?

> // Prepare component tags.
> component := []any{
> 	slog.String("component", "gRPCServer"),
> 	slog.String("interceptor", "Auth"),
> }

I wonder why not use `[]slog.Attr` here?

> Allow to ignore false positive matches.

Doesn't `//nolint` work for you? Or do you run `sloglint` standalone?

**Comment by pavelpatrin** [comment]
Thanx for reply, I'm using it via `golangci-lint`. Looks like I have a `0.7.1` version of sloglint.

> I wonder why not use []slog.Attr here?

Do you mean like this?

```
	attrs := []slog.Attr{
		slog.String("a", "b"),
		slog.String("c", "d"),
	}

	slog.Default().Info("Test", attrs...)
```

It is impossible, since go does not support _container covariance_. We can't pass `[]slog.Attr` if function accepts `[]any`.
Go have https://github.com/golang/go/issues/66937 issue where author asks to add `logger.WithAttrs(attrs...)`, but it is not ready.

> Doesn't //nolint work for you? Or do you run sloglint standalone?

`//nolint` helps, thank you very much. Didn't know that golangci-lint adds it's own ignorers.

**Comment by tmzane** [comment]
> Looks like I have a 0.7.1 version of sloglint.

I see, we should wait for `golangci-lint` to pick `0.7.2` then.

> It is impossible, since go does not support container covariance. We can't pass []slog.Attr if function accepts []any.

Yeah, my bad, I thought the call was `LogAttr` for some reason.

In theory, it should be possible to check simple cases, like the one you provided. In practice, `attrs` slices are usually used to log some fields conditionally, so we would need to check all `append` calls. Considering it's only useful for the `attr-only` mode, I'm not sure the complexity of the implementation is worth it.

**Comment by pavelpatrin** [comment]
Looks like 0.7.2 will fix my problem and you suggested to use `//nolint`, so thank you very much. I think this could be closed.
